File: tests/cases/basic/ternary-effects.ts
function statementPos(cond: boolean) {
  let i = 0;
  cond ? i++ : i--;
  return i;
}

function effectfulTrue(cond: boolean) {
  let i = 0;
  const r = cond ? (i += 1) : i;
  return r + i;
}

function effectfulFalse(cond: boolean) {
  let i = 0;
  const r = cond ? i : (i += 2);
  return r + i;
}

function pure(cond: boolean) {
  return cond ? 1 : 2;
}

File: tests/cases/basic/__snapshots__/ternary-effects.lua
local ____exports = {}
local function statementPos(cond)
    local i = 0
    if cond then
        i = i + 1
    else
        i = i - 1
    end
    return i
end
local function effectfulTrue(cond)
    local i = 0
    local ____ternary_0
    if cond then
        i = i + 1
        ____ternary_0 = i
    else
        ____ternary_0 = i
    end
    local r = ____ternary_0
    return r + i
end
local function effectfulFalse(cond)
    local i = 0
    local ____ternary_1
    if cond then
        ____ternary_1 = i
    else
        i = i + 2
        ____ternary_1 = i
    end
    local r = ____ternary_1
    return r + i
end
local function pure(cond)
    return (function()
        if cond then
            return 1
        else
            return 2
        end
    end)()
end
return ____exports
